src/control/modules/motion-control/PidMotionController.hpp

```diff
@@ -56,14 +102,85 @@ class PidMotionController {
                                float dt, Eigen::Vector4d* errors = nullptr,
                                Eigen::Vector4d* wheelVelsOut = nullptr,
                                Eigen::Vector4d* targetWheelVelsOut = nullptr) {
-        // convert encoder ticks to rad/s
+        // update control targets
+        // in the future, we can get the rotation angle soccer wants and
+        // directly command that
+        // as our target, or integrate the rotational velocities given to us to
+        // create the target.
+        // For now though, we only do an angle hold when soccer is not
+        // commanding rotational velocities
+        // (this should help with strafing quickly)
+        // target_rotation += _targetVel[2] * dt;
+
+        // get sensor data
+        imu.getMotion6(&ax, &ay, &az, &gx, &gy, &gz);
+
+        // convert sensor readings to mathematically valid values
         Eigen::Vector4d wheelVels;
         wheelVels << encoderDeltas[0], encoderDeltas[1], encoderDeltas[2],
             encoderDeltas[3];
         wheelVels *= 2.0 * M_PI / ENC_TICKS_PER_TURN / dt;
 
+        auto bot_vel = RobotModel::get().WheelToBot * wheelVels;
+
+        // we have two redundent sensor measurements for rotation
+        // 32.8 comes from data sheet, units are LSB / (deg/s)
+        float ang_vel_gyro = (gz / 32.8f) * M_PI / 180.0f;
```

Comment:
The typical scale factors are usually ok for the Gyro/Accelerometers, but if it becomes a problem, we can do some calibration to estimate those too. It's a pretty straightforward process